### ISSUE TITLE
Items_list parser

### ISSUE THREAD

**Issue by moneyisgod** [OpenKore/openkore#2168]
![image](https://user-images.githubusercontent.com/12580473/42904754-59ca6302-8aac-11e8-8408-911a074b800e.png)

As i now, first value of comparison will always be true ?! 
Is it intended to work at this way ?

**Comment by Nipodemos** [comment]
can you test it?
and i think it is `eq` instead of `==`
$args->{getter} is an object, i don't know what have to be used

**Comment by c4c1n6kr3m1** [comment]
@moneyisgod
i am pretty sure, the reason of issue https://github.com/OpenKore/openkore/issues/446 is really
```
$char->storage->clear
```
he got two 0995 packet

**Comment by Nipodemos** [comment]
now that you've put that way, i think it is that too @c4c1n6kr3m1 
it is just to initialize de object, should be changed

about the other issue i don't know enough to say something

**Comment by moneyisgod** [comment]
Yeah but how about clear storage after every session? We must clear when we open, not when close. We have the storage log function.  Very useful.

**Comment by moneyisgod** [comment]
If openkore now is detecting packet len not only in recvpackets, i guess it auto splits uh?

**Comment by c4c1n6kr3m1** [comment]
your point?

**Comment by moneyisgod** [comment]
I keep thinking it will.replace every first item of every category. Stack inv cart storage unst too

**Comment by c4c1n6kr3m1** [comment]
damn i have bad english, i am not on the same page
did u try to said that it's not only stack storage, but applied on stack inv and stack cart ?
can we not add a timeout  for that?

**Comment by moneyisgod** [comment]
Depending on scenario, if he receives twice packets for storage is doing this, maybe twice of inventory will do it too.

**Comment by c4c1n6kr3m1** [comment]
we do not clear inventory before the stackable inv , right?

**Comment by moneyisgod** [comment]
Oh now u told yea only in map change. O.o

**Comment by c4c1n6kr3m1** [comment]
the problem for me is he didnot give a screenshot
and his example is different with the example packet and wxstart

**Comment by moneyisgod** [comment]
Man is the second time i see this, one with me and one with him. 
It really only applied to sT, but will break all logs of storage?

Look my conspiration theory .

4144 also have several conspiration theory like that kore dont give u your realtime position.
How to fix these?

**Comment by c4c1n6kr3m1** [comment]
Maybe You're Right. more complicated

**Comment by moneyisgod** [comment]
Or maybe not jajaja i dont have even a computer now to do ir ahu3

**Comment by Nipodemos** [comment]
this bug of missing first item happen on official servers too? i can test it if you guys want

**Comment by moneyisgod** [comment]
Store all your items and keep in your inventory only one of stackable, and only one of unstackable.
Then try to have two stackable and only one unstackable. 

It will be sufficient to see if in private this error happens.

**Comment by lututui** [comment]
> As i know, first value of comparison will always be true ?!

That is not the case at all.
Basically, `$local_item = $args->{getter}` in perl assigns `$local_item` and **returns the assigned value**. If `$local_item = $args->{getter}` is a false value (undef, 0 or empty string) `$local_item = $args->{getter}` is false

Example:
![image](https://user-images.githubusercontent.com/17522256/43974208-37701ea8-9cb0-11e8-95fd-1663be3aaa28.png)

`unless ($local_item = $args->{getter} && $args->{getter}($item)) {` tests if `$args->{getter}` is any not-false value while assigning it to `$local_item` and then calls `$args->{getter}($item)`, ensuring we're not calling `undef($item)`. It could be rewritten as 

```
$local_item = $args->{getter};
unless ($local_item && $args->{getter}($item)) {
```

There's nothing wrong there.
